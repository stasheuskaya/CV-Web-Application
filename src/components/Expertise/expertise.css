.expertiseContainer {
    border-collapse: collapse;
    width: 100%;
    text-align: left;
  }
  
  .expertiseCompany, .expertiseJob,  .educationTitle, .educationDate  {
    font-weight: 700;
    font-size: 16px;
    line-height: 120%;
  }
  
  .expertiseCompany, .expertiseJob {
    padding-bottom: 5px;
  }
  
  .expertiseDate, .expertiseDescription {
    padding-bottom: 20px;
  }
  
  .expertiseCompany {
    width: 140px;
  }
  
  .expertiseJob, .expertiseDescription {
    padding-left: 30px;
  }

  .feedbackContainer {
    width: 100%;
  }
  