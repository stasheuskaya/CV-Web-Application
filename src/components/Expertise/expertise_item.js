export default function ExpertiseItem (props) {
    return (
        <table className="expertiseContainer">
            <tr>
                <th className="expertiseCompany">
                    {props.company}
                </th>
                <th className="expertiseJob">
                    {props.job}
                </th>
            </tr>
            <tr>
                <th className="expertiseDate">
                    {props.date}
                </th>
                <th className="expertiseDescription">
                    {props.description}
                </th>
            </tr>
        </table>
    )
  }
