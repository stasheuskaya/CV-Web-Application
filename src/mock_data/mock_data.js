export const expertisesData = [
  {
      date: '2021-2022', 
      info: {
        company: 'EPAM Systems inc',
        job: 'FrontEnd Developer/Engineer',
        description: 'Develop various web application, including the a Capstone project "CV Web application".  CV web application is a personal, professional CV web application that allows users to have a working prototype to share with audiences to introduce themselves.'
      }
  },
  {
      date: '2019-2021', 
      info: {
        company: 'Selfemployee',
        job: 'Mobile Software Testing Engineer  / QA',
        description: 'MyBudget mobile application (iOS). Set a budget and keep it with real time budget stats.'
      }
  },
  {
    date: '2017-2018', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Mobile Software Testing Engineer  / QA',
      description: 'Ecommerce mobile application (iOS, Android). Global specialist in energy management and automation, with global presence in over 100 countries. It provides integrated efficiency solutions, combining energy, automation and software.'
    }
  },
  {
    date: '2015-2016', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Software Testing Engineer',
      description: 'Ecommerce mobile application (iOS, Android) online-fashion outlet, with operations across Spain, Italy, Brazil and Mexico.'
    }
  },
  {
      date: '2014-2015', 
      info: {
        company: 'EPAM Systems inc',
        job: 'Software Testing Engineer',
        description: 'SaaS based Supply Chain and Industry platform, provides a set of Internet services for the automotive and manufacturing industry, with global presents in over 70 countries.'
      }
    }
]

export const feedbackData = [
  {feedback: '"Yulia is an incredibly talented, hard-working individual. I would recommend her to work in any area she pursues. She volunteered under me and also worked for a school I was on the board for. In both circumstances, paid and unpaid, she was dedicated, faithful, consistent, honest, kind, loving, full of good ideas, took initiative, and easy to work with."', 
  reporter: { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Allyse Tanovitz, YVCS School board member', 
  citeUrl: 'https://yvcschool.org' } 
  }, 

  {feedback: '"Yuliya was a valuable member of our team, and her job and attitude helped us to finish project successfully. Yuliya proved herself not only as a good QA engineer, but also as a consistent and reliable person, it was easy and comfortable to work with Yuliya."'
  , 
  reporter: 
  { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Marina Tretjakova, QA Test Lead, EPAM Systems', 
  citeUrl: 'www.epam.com' } } 
]

export const timeLineData = [
{
  "date": 2022,
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "date": 2015,
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "date": 2014,
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "date": 2008,
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const portfolioData = [
{
  "type": "ui",
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "type": "code",
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "type": "ui",
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "type": "code",
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const skillsData = [
{
  "name": "React",
  "range": 10
},
{
  "name": "JS",
  "range": 100
}
]