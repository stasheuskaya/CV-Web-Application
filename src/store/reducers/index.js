import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;