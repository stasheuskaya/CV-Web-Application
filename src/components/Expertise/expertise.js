import ExpertiseItem from "./expertise_item";
import './expertise.css';

export default function Expertise(props) {
    return (
      <div>
        {props.data.map(item => 
        <ExpertiseItem 
          date={item.date}
          key={item.date}
          company={item.info.company} 
          job={item.info.job} 
          description={item.info.description} 
          />)}
      </div>
    )
}
