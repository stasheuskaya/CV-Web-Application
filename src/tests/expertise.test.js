import renderer from "react-test-renderer";
import Expertise from "../components/Expertise/expertise";
import { Provider } from 'react-redux';
import { store } from "../store/store"
import { expertisesData } from "../mock_data/mock_data"

describe('TimeLine Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <Expertise data={expertisesData}/>
			</Provider>)).toMatchSnapshot();
	});
});