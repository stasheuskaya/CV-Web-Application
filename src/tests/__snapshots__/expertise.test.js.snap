// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLine Page renders content 1`] = `
<div>
  <table
    className="expertiseContainer"
  >
    <tr>
      <th
        className="expertiseCompany"
      >
        EPAM Systems inc
      </th>
      <th
        className="expertiseJob"
      >
        FrontEnd Developer/Engineer
      </th>
    </tr>
    <tr>
      <th
        className="expertiseDate"
      >
        2021-2022
      </th>
      <th
        className="expertiseDescription"
      >
        Develop various web application, including the a Capstone project "CV Web application".  CV web application is a personal, professional CV web application that allows users to have a working prototype to share with audiences to introduce themselves.
      </th>
    </tr>
  </table>
  <table
    className="expertiseContainer"
  >
    <tr>
      <th
        className="expertiseCompany"
      >
        Selfemployee
      </th>
      <th
        className="expertiseJob"
      >
        Mobile Software Testing Engineer  / QA
      </th>
    </tr>
    <tr>
      <th
        className="expertiseDate"
      >
        2019-2021
      </th>
      <th
        className="expertiseDescription"
      >
        MyBudget mobile application (iOS). Set a budget and keep it with real time budget stats.
      </th>
    </tr>
  </table>
  <table
    className="expertiseContainer"
  >
    <tr>
      <th
        className="expertiseCompany"
      >
        EPAM Systems inc
      </th>
      <th
        className="expertiseJob"
      >
        Mobile Software Testing Engineer  / QA
      </th>
    </tr>
    <tr>
      <th
        className="expertiseDate"
      >
        2017-2018
      </th>
      <th
        className="expertiseDescription"
      >
        Ecommerce mobile application (iOS, Android). Global specialist in energy management and automation, with global presence in over 100 countries. It provides integrated efficiency solutions, combining energy, automation and software.
      </th>
    </tr>
  </table>
  <table
    className="expertiseContainer"
  >
    <tr>
      <th
        className="expertiseCompany"
      >
        EPAM Systems inc
      </th>
      <th
        className="expertiseJob"
      >
        Software Testing Engineer
      </th>
    </tr>
    <tr>
      <th
        className="expertiseDate"
      >
        2015-2016
      </th>
      <th
        className="expertiseDescription"
      >
        Ecommerce mobile application (iOS, Android) online-fashion outlet, with operations across Spain, Italy, Brazil and Mexico.
      </th>
    </tr>
  </table>
  <table
    className="expertiseContainer"
  >
    <tr>
      <th
        className="expertiseCompany"
      >
        EPAM Systems inc
      </th>
      <th
        className="expertiseJob"
      >
        Software Testing Engineer
      </th>
    </tr>
    <tr>
      <th
        className="expertiseDate"
      >
        2014-2015
      </th>
      <th
        className="expertiseDescription"
      >
        SaaS based Supply Chain and Industry platform, provides a set of Internet services for the automotive and manufacturing industry, with global presents in over 70 countries.
      </th>
    </tr>
  </table>
</div>
`;
